File: integration-tests-java/src/main/java/io/micronaut/jimmer/java/it/config/UUIDGenerator.java
package io.micronaut.jimmer.java.it.config;

import java.util.UUID;
import org.babyfish.jimmer.sql.meta.UserIdGenerator;

public class UUIDGenerator implements UserIdGenerator<UUID> {

    @Override
    public UUID generate(Class<?> entityType) {
        return UUID.randomUUID();
    }
}

File: integration-tests-kotlin/src/main/kotlin/io/micronaut/jimmer/kotlin/it/config/UUIdGenerator.kt
package io.micronaut.jimmer.kotlin.it.config

import org.babyfish.jimmer.sql.meta.UserIdGenerator
import java.util.UUID

class UUIdGenerator : UserIdGenerator<UUID> {
    override fun generate(entityType: Class<*>?): UUID = UUID.randomUUID()
}
